pallets/pools/src/lib.rs

```diff
@@ -587,18 +619,22 @@ pub mod pallet {
 
 		/// Update per-pool configuration settings.
 		///
-		/// This sets the minimum epoch length, epoch solution challenge
-		/// time, and maximum NAV age.
+		/// This updates the tranches of the pool,
+		/// sets the minimum epoch length, and maximum NAV age.
+		///
+		/// If no delay is required for updates and redemptions
+		/// don't have to be fulfilled, then this is executed
+		/// immediately. Otherwise, the update is scheduled
+		/// to be executed later.
 		///
 		/// The caller must have the `PoolAdmin` role in order to
 		/// invoke this extrinsic.
-		#[pallet::weight(T::WeightInfo::update())]
+		#[pallet::weight(T::WeightInfo::update_no_execution()
+			.max(T::WeightInfo::update_and_execute()))]
 		pub fn update(
 			origin: OriginFor<T>,
 			pool_id: T::PoolId,
-			min_epoch_time: u64,
-			challenge_time: u64,
-			max_nav_age: u64,
+			changes: PoolChanges<T::InterestRate>,
 		) -> DispatchResult {
```

Comment:
This should be updated to use a `DispatchResultWithPostInfo` and the return value updated from `()` to the appropriate weight for each code paths. Otherwise users will always be paying for execution even if we defer it.